### DISCUSSION TITLE
SQL syntax error when there is no error - SQL Error [156] [S0001]: Incorrect syntax near the keyword 'from'.

### DISCUSSION THREAD

**Discussion by AdrianWalkerr** [dbeaver/dbeaver#36310] [Q&A] [answered]
### Description

SQL Error [156] [S0001]: Incorrect syntax near the keyword 'from'.

Resulting from this query:
select *

--case when SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) % 3600 / 60 <10  
--then
--
--    CONCAT(
--        cast(SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) / 3600 as varchar), ':', 
--		concat('0' , cast((SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) % 3600 / 60) as varchar))	
--		)
--else
--    CONCAT(
--        SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) / 3600, ':', 
--		(SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) % 3600) / 60)	
--end

from [dbo].[FlightFollow]

where 
[DepartureScheduled] > '2024-11-10 00:00:00' and [DepartureScheduled] < '2024-11-18 23:00:00' -- change the months on this line. Date is in YYYY-MM-DD HH:MM:SS format
and Deleted='0' 
and dbo.FlightFollow.Id is not null 
and ActualRouteId is not null 
and Cancelled='0' 
and ActualFromPortId is not null
and GroundReturn='0'
 

### DBeaver Version

CE 24.2.5

### Operating System

Mint 22

### Database and driver

DB: SQL Server 12.0
Driver: Microsoft JDBC Driver for SQL Server

### Steps to reproduce

1. Put the below query into the editor
select *

--case when SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) % 3600 / 60 <10  
--then
--
--    CONCAT(
--        cast(SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) / 3600 as varchar), ':', 
--		concat('0' , cast((SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) % 3600 / 60) as varchar))	
--		)
--else
--    CONCAT(
--        SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) / 3600, ':', 
--		(SUM(DATEDIFF(SECOND, DepartureBest, ArrivalBest)) % 3600) / 60)	
--end

from [dbo].[FlightFollow]

where 
[DepartureScheduled] > '2024-11-10 00:00:00' and [DepartureScheduled] < '2024-11-18 23:00:00' -- change the months on this line. Date is in YYYY-MM-DD HH:MM:SS format
and Deleted='0' 
and dbo.FlightFollow.Id is not null 
and ActualRouteId is not null 
and Cancelled='0' 
and ActualFromPortId is not null
and GroundReturn='0'
 
2. Observe error.
3. Note that this query works in MS SSMS and passes validation in VS Code as well.

**Comment by E1izabeth** [comment]
Could you do the following actions? It will help us understanding your issue
1. Open Query Manager
![image](https://github.com/user-attachments/assets/f505f676-a578-4af2-a786-ba6c1273c17f)

2. Turn on showing of User, User Script types of queries 
![image](https://github.com/user-attachments/assets/8065c206-dad8-4457-b597-daeb1c2f0616)

3. Clear log
![image](https://github.com/user-attachments/assets/c7d94ecb-83d9-4684-8862-01ffb182a39e)

3. Reproduce the issue once again

4.  Check if the query you see in the query manager equals to what you expected

**Comment by E1izabeth** [comment]
Also could you try select whole query and then press Run? Is it still gives the error if you execute this way?

**Comment by AdrianWalkerr** [comment]
Ahhh, thank you!  When I select the whole query and press Run it executes successfully.  Also, since I'm very new to DBeaver I had never used the Execute script button.  If I use that it executes successfully too (with the extra new lines in the query).  

I discovered is that if I remove all the blank lines it executes successfully using the run query button.

**Comment by E1izabeth** [comment, marked as answer]
By default, DBeaver treats a new line as a statement separator. You can change this behavior in `Preferences->Editors->SQL Editor->SQL Processing->Blank line is a statement delimiter`. Does it solve the issue?
